components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java

```diff
@@ -1798,6 +1823,121 @@ private void parseSupportedGrantTypesConfig(OMElement oauthConfigElem) {
+            }
+        }
+
+        //Adding default token types if not added in the configuration
+        if (!supportedTokenIssuers.containsKey(DEFAULT_TOKEN_TYPE)) {
+            supportedTokenIssuers.put(DEFAULT_TOKEN_TYPE,
+                    new TokenIssuerDO(DEFAULT_TOKEN_TYPE, DEFAULT_OAUTH_TOKEN_ISSUER_CLASS, true));
+        }
+        if (!supportedTokenIssuers.containsKey(JWT_TOKEN_TYPE)) {
+            supportedTokenIssuers.put(JWT_TOKEN_TYPE, new TokenIssuerDO(JWT_TOKEN_TYPE, JWT_TOKEN_ISSUER_CLASS, true));
+        }
+
+        boolean isRegistered = false;
+        //Adding global token issuer configured in the identity xml as a supported token issuer
+        for (Map.Entry<String, TokenIssuerDO> entry : supportedTokenIssuers.entrySet()) {
+            TokenIssuerDO issuerDO = entry.getValue();
+            if (oauthIdentityTokenGeneratorClassName != null && oauthIdentityTokenGeneratorClassName
+                    .equals(issuerDO.getTokenImplClass())) {
+                isRegistered = true;
+                break;
+            }
+        }
+
+        if (!isRegistered && oauthIdentityTokenGeneratorClassName != null) {
+            boolean isPersistTokenAlias = true;
+            if (persistAccessTokenAlias != null) {
+                isPersistTokenAlias = Boolean.valueOf(persistAccessTokenAlias);
+            }
+            supportedTokenIssuers.put(oauthIdentityTokenGeneratorClassName,
+                    new TokenIssuerDO(oauthIdentityTokenGeneratorClassName, oauthIdentityTokenGeneratorClassName,
+                            isPersistTokenAlias));
+        }
+    }
+
+    /**
+     * Adds oauth token issuer instances used for token generation.
+     * @param tokenType registered token type
+     * @return token issuer instance
+     * @throws IdentityOAuth2Exception
+     */
+    public OauthTokenIssuer addAndReturnTokenIssuerInstance(String tokenType) throws IdentityOAuth2Exception {
+
+        TokenIssuerDO tokenIssuerDO = supportedTokenIssuers.get(tokenType);
+        OauthTokenIssuer oauthTokenIssuer = null;
+        if (tokenIssuerDO != null && tokenIssuerDO.getTokenImplClass() != null) {
+            try {
+                if (oauthTokenIssuerMap.get(tokenType) == null) {
+                    Class clazz = this.getClass().getClassLoader().loadClass(tokenIssuerDO.getTokenImplClass());
+                    oauthTokenIssuer = (OauthTokenIssuer) clazz.newInstance();
+                    oauthTokenIssuer.setPersistAccessTokenAlias(
+                            supportedTokenIssuers.get(tokenType).isPersistAccessTokenAlias());
+                    oauthTokenIssuerMap.put(tokenType, oauthTokenIssuer);
+                    log.info("An instance of " + tokenIssuerDO.getTokenImplClass()
+                            + " is created for Identity OAuth token generation.");
+                } else {
+                    oauthTokenIssuer = oauthTokenIssuerMap.get(tokenType);
+                }
+            } catch (Exception e) {
```

Comment:
Is there a reason to catch generic Exception here?

Reply:
These are used to catch for class loading and instantiating issues. Similar implementations can be found in the same class such as https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java#L958